examples/server/server.cpp

```diff
@@ -131,9 +131,22 @@ struct slot_params {
             lora.push_back({{"id", i}, {"scale", this->lora[i].scale}});
         }
 
-        std::vector<std::string> grammar_trigger_words;
-        for (const auto & trigger : sampling.grammar_trigger_words) {
-            grammar_trigger_words.push_back(trigger.word);
+        auto grammar_triggers = json::array();
```

Comment:
Could `grammar_trigger` be a struct instead of json?

```cpp
struct grammar_trigger {
  common_grammar_trigger type; // enum?
  llama_token val_token;
  std::string val_string;

  // optionally, implement to_json() and from_json(json & input)
}
```

Reply:
added json serialization to common_grammar_trigger w/ the same template trick as in chat.h to avoid a hard dep to json.